contracts/hooks/examples/FullRange.sol

```diff
@@ -295,7 +299,7 @@ contract FullRange is BaseHook, ILockCallback {
         pool.hasAccruedFees = false;
     }
 
-    function lockAcquired(bytes calldata rawData)
+    function lockAcquired(address, /*sender*/ bytes calldata rawData)
```

Comment:
I think this contract now has a critical vuln in it.
Now that anyone can call `manager.lock()` with this hook as the locker and they as the lockCaller, they can pass in any data that they want. This means that `data.sender` can no longer be trusted. I think someone could call and get thisfunction to remove liquidity from the pool, and then send it to themselves.
I think we should no longer be using `data.sender` as a model, but instead be using `lockCaller` to find the sender

Reply:
also separately i think some of the functions in this hook need the `onlyPoolManager` modifier - e.g. `beforeSwap` doesnt have it?

Reply:
yeah good point. I think we can check that sender == address(this) and that the lock was started by the hook itself